### PR TITLE
feat(pt): support loss plugin for external package

### PR DESCRIPTION
**Pull request by ChiahsinChu**
support loss plugin for external package in pt backend

## Summary by CodeRabbit

- **New Features**
	- Introduced a new loss class, `TaskLoss`, enhancing loss functionality in training.
	- Updated loss selection process to dynamically accommodate new loss types.

- **Bug Fixes**
	- Improved error handling for unsupported loss types, allowing for more flexible integration.
	- Integrated plugin registry functionality into the `TaskLoss` class for better registration within the system.

### CODE REVIEW

File: deepmd/pt/train/training.py [hunk 1]
```diff
@@ -1260,7 +1261,8 @@ def get_loss(loss_params, start_lr, _ntypes, _model):
         loss_params["task_dim"] = task_dim
         return PropertyLoss(**loss_params)
     else:
-        raise NotImplementedError
+        loss_params["starter_learning_rate"] = start_lr
+        return TaskLoss.get_class_by_type(loss_type).get_loss(loss_params)
```
Comment (njzjz) [inline deepmd/pt/train/training.py:1265]: There is no `get_loss` defined in the base class.

**Review by njzjz — CHANGES_REQUESTED** [review]

**Review by njzjz — APPROVED** [review]
